actors/init/src/testing.rs

```diff
@@ -0,0 +1,63 @@
+use std::collections::HashMap;
+
+use fil_actors_runtime::{Map, MessageAccumulator, FIRST_NON_SINGLETON_ADDR};
+use fvm_ipld_blockstore::Blockstore;
+use fvm_shared::{
+    address::{Address, Protocol},
+    ActorID,
+};
+
+use crate::State;
+
+pub struct StateSummary {
+    pub ids_by_address: HashMap<Address, ActorID>,
+    pub next_id: ActorID,
+}
+
+// Checks internal invariants of init state.
+pub fn check_state_invariants<BS: Blockstore>(
+    state: &State,
+    store: &BS,
+) -> (StateSummary, MessageAccumulator) {
+    let acc = MessageAccumulator::default();
+
+    acc.require(!state.network_name.is_empty(), "network name is empty");
+    acc.require(
+        state.next_id >= FIRST_NON_SINGLETON_ADDR,
+        format!("next id {} is too low", state.next_id),
+    );
+
+    let mut init_summary = StateSummary { ids_by_address: HashMap::new(), next_id: state.next_id };
+
+    let mut address_by_id = HashMap::<ActorID, Address>::new();
+    match Map::<_, ActorID>::load(&state.address_map, store) {
+        Ok(address_map) => {
+            let ret = address_map.for_each(|key, actor_id| {
+                let key_address = Address::from_bytes(key)?;
+
+                acc.require(
+                    key_address.protocol() != Protocol::ID,
+                    format!("key {key_address} is an ID address"),
+                );
```

Comment:
missing 
```rust
acc.require(key_address.protocol() <= Protocol::BLS, format!("key {key_address} has unknown address protocol {}", key_address.protocol()));
```

Reply:
@ZenGround0 It's not really missing in Rust, see the [underlying enum](https://github.com/filecoin-project/ref-fvm/blob/dd70273521fe4a14e47fc92f3df4a5d4fcae4919/shared/src/address/protocol.rs#L13). In this case it cannot have an invalid state. Or am I missing something?